test cases/common/15 mixed pch/meson.build

```diff
@@ -3,4 +3,4 @@ project('mixed C and C++ pch test', 'cpp', 'c')
 
 exe = executable('prog', 'main.cc', 'func.c',
 c_pch : ['pch/func.h', 'pch/func_pch.c'],
-cpp_pch : ['pch/main_pch.cc', 'pch/main.h'])
+cpp_pch : ['pch/main_pch.cc', 'pch/main.hh'])
```

Comment:
The commit message explains this — it seemed to make it more likely that Intel will decide that the header is to be precompiled as a C++ header. I don't know if we can do something to help Intel decide that. I will investigate.